.github/workflows/dependabot.yml

```diff
@@ -0,0 +1,41 @@
+# Dependabot files PRs by pushing branches named
+# "dependabot/{ecosystem}/{depname}".  What it pushes to those
+# branches is pretty naive; it doesn't support us telling it that it
+# needs to run "make generate" after making an edit.
+#
+# But we can work around that by creating a workflow run "make
+# generate" on those branches!
+name: dependabot
+"on":
+  push:
+    branches:
+      - 'dependabot/**'
+
+jobs:
+  repatriate:
+    runs-on: ubuntu-latest
+    name: repatriate
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install Deps
+        uses: ./.github/actions/setup-deps
+      - name: Log in to GitHub
+        shell: bash
+        run: |
+          git config --global user.name '${{ secrets.GH_AUTO_USER }}'
+          git config --global user.email '${{ secrets.GH_AUTO_EMAIL }}'
+          git config --global 'url.https://${{ secrets.GH_GITHUB_API_KEY }}@github.com.insteadof' https://github.com
+      - run: |
+          make go-mod-tidy
+          make generate
+
+          git add .
+          if test -z "$(git status -s)"; then
+            # already up-to-date
+            exit 0
+          fi
+
+          git commit --signoff --message='make go-mod-tidy && make generate'
+          git push origin
```

Comment:
if the workflow is triggered on a push that doesn't cause `actions/checkout` to checkout in detached HEAD state using the commit that was just pushed?

Reply:
Good question.

Reply:
I've changed it to push `HEAD:${{ github.ref }}` so that it works either way.

Reply:
Looking at the CI run - yup looks it does checkout in detached HEAD state (good to know for the future)